fastcrypto-zkp/src/bn254/poseidon.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use ark_bn254::Fr;
+use light_poseidon::Poseidon;
+use light_poseidon::PoseidonHasher;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+/// Wrapper struct for Poseidon hash instance.
+pub struct PoseidonWrapper {
+    instance: Poseidon<Fr>,
+}
+
+impl Debug for PoseidonWrapper {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PoseidonWrapper").finish()
+    }
+}
+
+impl PoseidonWrapper {
+    /// Initialize a Poseidon hash function with the given size.
+    pub fn new(size: usize) -> Self {
+        Self {
+            instance: Poseidon::<Fr>::new_circom(size).unwrap(),
+        }
+    }
+
+    /// Calculate the hash of the given inputs.
+    pub fn hash(&mut self, inputs: &[Fr]) -> Fr {
+        self.instance.hash(inputs).unwrap()
+    }
+}
+#[cfg(test)]
+mod test {
+    use crate::bn254::zk_login::calculate_merklized_hash;
+
+    use super::PoseidonWrapper;
+    use ark_bn254::Fr;
+    use std::str::FromStr;
+    #[test]
+    fn poseidon_test() {
```

Comment:
I think it might be a good idea to be more robust with our Poseidon tests since we are using new libraries. 

We could generate 15 test vectors, one for each vector length, i.e., 1, 2, 3, ..., 15, and use these to make sure that all the libraries (Rust's light_poseidon, circom's internal one and JS's poseidon-light) conform with each other.

Reply:
adding todo for now